### ISSUE TITLE
Erro module 'fs/jbd2/commit.o' on function '@jbd2_submit_inode_data'

### ISSUE THREAD

**Issue by ViNi-Arco** [ClangBuiltLinux/linux#2008]
Hi peep,

When building Kernel 6.6.21 with CLANG/LLVM 17.0.6, I get this error:

```c
**Full LTO**

PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Program arguments: ld.lld -m elf_x86_64 -mllvm -import-instr-limit=5 -z noexecstack -r -o fs/jbd2/jbd2.o @fs/jbd2/jbd2.mod
1.      Running pass 'Function Pass Manager' on module 'ld-temp.o'.
2.      Running pass 'Prologue/Epilogue Insertion & Frame Finalization' on function '@jbd2_submit_inode_data'
 #0 0x00007fd83a2d74ab llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/usr/lib/libLLVM-17.so+0xa784ab)
 #1 0x00007fd83a2d5724 llvm::sys::RunSignalHandlers() (/usr/lib/libLLVM-17.so+0xa76724)
 #2 0x00007fd83a2d5886 (/usr/lib/libLLVM-17.so+0xa76886)
 #3 0x00007fd839329770 (/usr/lib/libc.so.6+0x40770)
 #4 0x00007fd83a7e3942 (/usr/lib/libLLVM-17.so+0xf84942)
 #5 0x00007fd83a7e99df (/usr/lib/libLLVM-17.so+0xf8a9df)
 #6 0x00007fd83a6c1098 (/usr/lib/libLLVM-17.so+0xe62098)
 #7 0x00007fd83a41668a llvm::FPPassManager::runOnFunction(llvm::Function&) (/usr/lib/libLLVM-17.so+0xbb768a)
 #8 0x00007fd83a416883 llvm::FPPassManager::runOnModule(llvm::Module&) (/usr/lib/libLLVM-17.so+0xbb7883)
 #9 0x00007fd83a417130 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/usr/lib/libLLVM-17.so+0xbb8130)
#10 0x00007fd83bb17a1e (/usr/lib/libLLVM-17.so+0x22b8a1e)
#11 0x00007fd83bb181ff llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/usr/lib/libLLVM-17.so+0x22b91ff)
#12 0x00007fd83bb0e088 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>) (/usr/lib/libLLVM-17.so+0x22af088)
#13 0x00007fd83bb0e451 llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>> (unsigned int, llvm::StringRef, llvm::Twine const&)>) (/usr/lib/libLLVM-17.so+0x22af451)
#14 0x0000561a7515b78c lld::elf::BitcodeCompiler::compile() (/usr/bin/ld.lld+0x23b78c)
#15 0x0000561a750dcadf void lld::elf::LinkerDriver::compileBitcodeFiles<llvm::object::ELFType<(llvm::support::endianness)1, true>>(bool) (/usr/bin/ld.lld+0x1bcadf)
#16 0x0000561a750e3c71 lld::elf::LinkerDriver::link(llvm::opt::InputArgList&) (/usr/bin/ld.lld+0x1c3c71)
#17 0x0000561a750e6a0d lld::elf::LinkerDriver::linkerMain(llvm::ArrayRef<char const*>) (/usr/bin/ld.lld+0x1c6a0d)
#18 0x0000561a750e842c lld::elf::link(llvm::ArrayRef<char const*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool) (/usr/bin/ld.lld+0x1c842c)
#19 0x0000561a74ff507d lld::unsafeLldMain(llvm::ArrayRef<char const*>, llvm::raw_ostream&, llvm::raw_ostream&, llvm::ArrayRef<lld::DriverDef>, bool) (/usr/bin/ld.lld+0xd507d)
#20 0x0000561a74ff49f6 lld_main(int, char**, llvm::ToolContext const&) (/usr/bin/ld.lld+0xd49f6)
#21 0x0000561a74ff33ef main (/usr/bin/ld.lld+0xd33ef)
#22 0x00007fd839312cd0 (/usr/lib/libc.so.6+0x29cd0)
#23 0x00007fd839312d8a __libc_start_main (/usr/lib/libc.so.6+0x29d8a)
#24 0x0000561a74ff4215 _start (/usr/bin/ld.lld+0xd4215)
make[4]: *** [scripts/Makefile.build:438: fs/jbd2/jbd2.o] Error 139
make[3]: *** [scripts/Makefile.build:480: fs/jbd2] Error 2


**Thin LTO**

PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Running pass 'Function Pass Manager' on module 'fs/jbd2/commit.o'.
1.      Running pass 'Prologue/Epilogue Insertion & Frame Finalization' on function '@jbd2_submit_inode_data'
LLVM ERROR: Failed to rename temporary file .thinlto-cache/Thin-fbedce.tmp.o to .thinlto-cache/llvmcache-5F97E09C5D7973571064E5019CEFD2027FA4E250: No such file or directory
```

**Comment by ViNi-Arco** [comment]
This too

```c
  LD [M]  arch/x86/crypto/ghash-clmulni-intel.o
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Program arguments: ld.lld -m elf_x86_64 -mllvm -import-instr-limit=5 -z noexecstack -r -o arch/x86/crypto/ghash-clmulni-intel.o @arch/x86/crypto/ghash-clmulni-intel.mod
1.      Running pass 'Function Pass Manager' on module 'ld-temp.o'.
2.      Running pass 'Prologue/Epilogue Insertion & Frame Finalization' on function '@ghash_async_init'
 #0 0x00007fca641d34ab llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/usr/lib/libLLVM-17.so+0xa784ab)
 #1 0x00007fca641d1724 llvm::sys::RunSignalHandlers() (/usr/lib/libLLVM-17.so+0xa76724)
 #2 0x00007fca641d1886 (/usr/lib/libLLVM-17.so+0xa76886)
 #3 0x00007fca63225770 (/usr/lib/libc.so.6+0x40770)
 #4 0x00007fca646df942 (/usr/lib/libLLVM-17.so+0xf84942)
 #5 0x00007fca646e59df (/usr/lib/libLLVM-17.so+0xf8a9df)
 #6 0x00007fca645bd098 (/usr/lib/libLLVM-17.so+0xe62098)
 #7 0x00007fca6431268a llvm::FPPassManager::runOnFunction(llvm::Function&) (/usr/lib/libLLVM-17.so+0xbb768a)
 #8 0x00007fca64312883 llvm::FPPassManager::runOnModule(llvm::Module&) (/usr/lib/libLLVM-17.so+0xbb7883)
 #9 0x00007fca64313130 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/usr/lib/libLLVM-17.so+0xbb8130)
#10 0x00007fca65a13a1e (/usr/lib/libLLVM-17.so+0x22b8a1e)
#11 0x00007fca65a141ff llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/usr/lib/libLLVM-17.so+0x22b91ff)
#12 0x00007fca65a0a088 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>) (/usr/lib/libLLVM-17.so+0x22af088)
#13 0x00007fca65a0a451 llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>> (unsigned int, llvm::StringRef, llvm::Twine const&)>) (/usr/lib/libLLVM-17.so+0x22af451)
#14 0x000055ed38b4078c lld::elf::BitcodeCompiler::compile() (/usr/bin/ld.lld+0x23b78c)
#15 0x000055ed38ac1adf void lld::elf::LinkerDriver::compileBitcodeFiles<llvm::object::ELFType<(llvm::support::endianness)1, true>>(bool) (/usr/bin/ld.lld+0x1bcadf)
#16 0x000055ed38ac8c71 lld::elf::LinkerDriver::link(llvm::opt::InputArgList&) (/usr/bin/ld.lld+0x1c3c71)
#17 0x000055ed38acba0d lld::elf::LinkerDriver::linkerMain(llvm::ArrayRef<char const*>) (/usr/bin/ld.lld+0x1c6a0d)
#18 0x000055ed38acd42c lld::elf::link(llvm::ArrayRef<char const*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool) (/usr/bin/ld.lld+0x1c842c)
#19 0x000055ed389da07d lld::unsafeLldMain(llvm::ArrayRef<char const*>, llvm::raw_ostream&, llvm::raw_ostream&, llvm::ArrayRef<lld::DriverDef>, bool) (/usr/bin/ld.lld+0xd507d)
#20 0x000055ed389d99f6 lld_main(int, char**, llvm::ToolContext const&) (/usr/bin/ld.lld+0xd49f6)
#21 0x000055ed389d83ef main (/usr/bin/ld.lld+0xd33ef)
#22 0x00007fca6320ecd0 (/usr/lib/libc.so.6+0x29cd0)
#23 0x00007fca6320ed8a __libc_start_main (/usr/lib/libc.so.6+0x29d8a)
#24 0x000055ed389d9215 _start (/usr/bin/ld.lld+0xd4215)
make[4]: *** [scripts/Makefile.build:438: arch/x86/crypto/ghash-clmulni-intel.o] Error 139
make[3]: *** [scripts/Makefile.build:480: arch/x86/crypto] Error 2
```

**Comment by nathanchance** [comment]
This is almost certainly a duplicate of #1984.

**Comment by ViNi-Arco** [comment]
> This is almost certainly a duplicate of https://github.com/ClangBuiltLinux/linux/issues/1984.

Thanks for the info, I'll test LLVM 17.0.6 with the patch later, if it fixes it.

CLANG/LLVM 15.0.7 built Kernel 6.6.21 without errors, with only these two warnings, but the Kernel is functional:

```c
LD [M]  drivers/gpu/drm/amd/amdgpu/amdgpu.o
drivers/gpu/drm/amd/amdgpu/amdgpu.o: warning: objtool: .text.navi10_umc_hybrid_cdr_workaround: unexpected end of section

vmlinux.o: warning: objtool: set_ftrace_ops_ro+0x37: relocation to !ENDBR: relocate_range+0x6d3
vmlinux.o: warning: objtool: mark_rodata_ro+0x11c: relocation to !ENDBR: relocate_range+0x6d3
```

**Comment by ViNi-Arco** [comment]
@nathanchance 

Yes! applying the [commit f800c1f](https://github.com/llvm/llvm-project/commit/f800c1f3b207e7bcdc8b4c7192928d9a078242a0) in llvm 17.0.6 fixed it, Kernel 6.6.21 built correctly..

Thanks Nathan

**Comment by nathanchance** [comment]
Thanks for testing!
